client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -723,11 +739,11 @@ class CalypsoifyIframe extends Component<
 							className={ isIframeLoaded ? 'is-loaded' : '' }
 							ref={ this.iframeRef }
 							src={ isIframeLoaded ? currentIFrameUrl : iframeUrl }
-							// Iframe url needs to be kept in state to prevent editor reloading if frame_nonce changes
-							// in Jetpack sites
-							onLoad={ () => {
-								this.onIframeLoaded( iframeUrl );
-							} }
+							// NOTE: Do not include any "editor load" events in
+							// this handler. It really only tracks if the document
+							// has loaded. Use the WindowActions.Loaded onMessage
+							// handler to detect when the editor has loaded.
+							onLoad={ () => this.setEditorRedirectTimer( 1000 ) }
```

Comment:
Any chance to optimize loading speed here by waiting only for DOM to load, and not CSS/other assets? See notes at https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload

Reply:
The reason this exists is to detect if the iframe loads a page, but it isn't the editor (e.g. 404 or unauthenticated page). In that case, it's not easy to detect those errors inside the iframe. So we're relying on the iframe bridge server post message to tell us when the editor page exists. In my testing, that takes a few hundred ms or so.

This wouldn't save any time on the happy path because all of the logic dealing with showing the editor happens after receiving the message from the iframe bridge server. And that's kind of how it has to be because the calypso integration won't work unless the bridge server is initialized.

At any rate, I don't _think_ we have been introducing any artificial delay. The redirect timer is always cleared immediately after we detect that the editor has loaded

Reply:
Gotcha, thanks for elaborating!